### DISCUSSION TITLE
API request is redirected to Cloudflare's real human checking page.

### DISCUSSION THREAD

**Discussion by butaixianran** [civitai/civitai#355] [General]
When sending a request to civitai API to get information, response code is 403 and content is the Cloudflare page for real human checking. 

But this is an official API route. It is not reasonable to check if the request is sent by real human. 

If civitai is worrying about the traffic from API route, you can ask for API key and set a delay between 2 requests for free user.

For example, some cloud service providers have a rule as "no more than 1 request in a second for free user".

I'm sure civitai team can find a better way for civitai. Anyway, redirect an API request to a real human checking page is not the right way.

**Comment by butaixianran** [comment]
ok, seems this is a temporary issue. But put API route under Cloudflare is not a good solution for anything. Let's hope you can find a better solution for this.

**Comment by aaron-dunamu** [comment]
If you add fake header like below, it will be work! I have added header to source code manually.
`HEADERS = {'User-Agent': 'Mozilla/5.0 (iPad; CPU OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148'}`

**Reply by butaixianran** [reply]
I don't think Cloudflare could be so easliy fooled with a header like this. If this is ture, then Cloudflare gonna lose trust from its clients and bankrupt immediately.  

But still thanks for your information. I'll check it out.

**Reply by WASasquatch** [reply]
> I don't think Cloudflare could be so easliy fooled with a header like this. If this is ture, then Cloudflare gonna lose trust from its clients and bankrupt immediately.
> 
> But still thanks for your information. I'll check it out.

It's more about what cloudfare serves, isn't a security issue. Just a content type trigger.

**Comment by butaixianran** [comment]
Ok, tried it, not sure if it works or not, but no harm anyway, so I added it. 
Still, if this is a trigger, that really means civitai should not put an API service under this rule.

**Reply by JustMaier** [reply]
Yeah, that's a bummer. I would have hoped that they would look at the content-type on the request and behaved accordingly but I guess not...
I've disabled the thing that was triggering this.

**Reply by butaixianran** [reply]
Good to know. A lot of civitai helper's users are redirected to Cloudflare and keeps claiming this extension is broken. I'm kind of tired of this.

**Reply by JustMaier** [reply]
Yeah, shouldn't be an issue anymore 👍

**Reply by butaixianran** [reply]
I think the real solution for this is one day, api need to move to a sub-domain, like api.civitai.com. Then it can has its own security rules.
